tools/lib/url_file.py

```diff
@@ -6,23 +6,36 @@
 import threading
 import urllib.parse
 import pycurl
+from hashlib import sha256
 from io import BytesIO
 from tenacity import retry, wait_random_exponential, stop_after_attempt
 
+#Cache chunk size
+K=1000
+CHUNK_SIZE=1000 * K
 
+PATH="/tmp/comma_download_cache/"
+LENGTHS_SUBFOLDER="lengths/" 
 class URLFile(object):
   _tlocal = threading.local()
 
-  def __init__(self, url, debug=False):
+  def __init__(self, url, debug=False, cache=True):
```

Comment:
- Cache needs to be off by default
 - Also have env variable to turn on caching, overriding this parameter